utils/getSetupMetadata.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable local-rules/require-data-mapper */
+import type { SetupMetadata } from '@prisma/client';
+import { prisma } from '~/utils/db';
+
+export default async function getSetupMetadata(): Promise<SetupMetadata> {
+  let setupMetadata = await prisma.setupMetadata.findFirst();
+
+  // if no setup metadata exists, seed it
+  if (!setupMetadata) {
+    setupMetadata = await prisma.setupMetadata.create({
+      data: {
+        id: 1,
+        configured: false,
+        initializedAt: new Date(),
+        onboarded: false,
+      },
+    });
+  }
+
+  return setupMetadata;
+}
```

Comment:
As I mentioned above, we need to call this on basically every request. It should be called in the root layout. Because of the perf implications of hitting the DB on every request, we should wrap either the whole function or the DB request in react `cache()`: https://react.dev/reference/react/cache

Reply:
- moved to root https://github.com/complexdatacollective/Fresco/pull/10/commits/eaf7ebd621cd640807955e8f7feaffd3276b5f42
- cache metadata db request https://github.com/complexdatacollective/Fresco/pull/10/commits/f3af5fdf81c08f43306acd2a4b6f247bee1eeb5b